### PR TITLE
dependsOn completions, part 1

### PR DESCRIPTION
**Pull request by StephenWeatherford**
Fixes most of #492

### CODE REVIEW

File: src/documents/positionContexts/TemplatePositionContext.ts [hunk 1]
```diff
@@ -366,11 +362,62 @@ export class TemplatePositionContext extends PositionContext {
...
+            }
+        }
+
+        return this.document.topLevelScope;
+    }
+
+    /**
+     * Gets the nearest resource object that contains the current position
+     */
+    public getEnclosingResource(): TemplateScope {
```
Comment (bhsubra) [inline src/documents/positionContexts/TemplatePositionContext.ts:393]: I don't see any difference between this and getScope(). Am I missing something?

File: src/documents/templates/getResourcesInfo.ts [hunk 2]
```diff
@@ -21,28 +22,139 @@ export function getResourcesInfo(scope: TemplateScope): IResourceInfo[] {
...
     }
+}
 
-    public get fullTypeName(): string {
-        if (this.typeExpressions.length > 1) {
-            return `'${this.typeExpressions.map(removeSingleQuotes).join('/')}'`;
+/**
+ * Concatenates a list of TLE expressions into a single expression, using 'concat' if necessary, and combining string literals when possible.
+ * @param expressions TLE expressions to concat
+ * @param unquotedLiteralSeparator An optional separator string literal (without the quotes) to place between every expresion
```
Comment (bhsubra) [inline src/documents/templates/getResourcesInfo.ts:102]: nit: expresion -> expression

File: src/vscodeIntegration/toVsCodeCompletionItem.ts [hunk 3]
```diff
@@ -110,6 +71,43 @@ export function toVsCodeCompletionItem(deploymentFile: DeploymentDocument, item:
...
+            return vscode.CompletionItemKind.Property;
+
+        case Completion.CompletionKind.PropertyValueForNewProperty:
+            return vscode.CompletionItemKind.Snippet;
+
+        case Completion.CompletionKind.tleResourceIdResTypeParameter:
+        case Completion.CompletionKind.tleResourceIdResNameParameter:
+            return vscode.CompletionItemKind.Reference;
+
+        case Completion.CompletionKind.dependsOnResourceId:
```
Comment (bhsubra) [inline src/vscodeIntegration/toVsCodeCompletionItem.ts:100]: Include/merge these in previous two cases?

File: src/documents/positionContexts/TemplatePositionContext.ts (hunk 1 above)
Comment (StephenWeatherford) [reply src/documents/positionContexts/TemplatePositionContext.ts:393]: You're right, this wasn't even used.

**Review by bhsubra — APPROVED** [review]

**Pull request merged**